### DISCUSSION TITLE
Confused Net vs Framework

### DISCUSSION THREAD

**Discussion by TobiasPSP** [vvvv/SVG#1142] [Q&A]
Spent days now figuring out DLL versions. No avail. I guess it's time now to ask for help from some experts:

What I want to do is use SvgDocument within a PowerShell script both in "Windows PowerShell" (.NET Framework) **and** "PowerShell 7" (NET).

```powershell
Add-Type -Path c:\test\svg.dll -PassThru
```

I am prepared to add logic to load different DLLs based on platform using `Add-Type` but would prefer a compatible DLL that works in both worlds as-is.

I did find a DLL version 3.4.6.1 that works well in "PowerShell 7" (NET) without any dependencies for what I need to do. 
It won't load in "Windows Framework" though.

I then found same version DLLs that sucessfully load in "Windows PowerShell" but lacks many classes like "SvgDocument".

I finally started a VisualStudio WPF Framework sample project and added SVG.NET via nuget. This again provided a namespace that started with "SVGNET" (as opposed to "SVG"), missing many classes such as "SvgDocument".

I *assumed* there is a common code base that is compatible with .NET Framework. Now I start to believe that the dll that is used with framework applications ("SVGNET") is actually a different beast with a different object model. Is that really so?

Sorry if my questions are naive, I tried a lot and spent much time to figure it out based on avail doc. 
Could someone point me into the right direction? Many thanks.

**Comment by mrbean-bremen** [comment]
> I finally started a VisualStudio WPF Framework sample project and added SVG.NET via nuget

SVG.NET is the name of the organization (_EDIT:_ actually "svg-net" is the name of the organization, got that wrong), not of the framework - that is just SVG (at least for this framework, maybe you are using another one). You may need to add SVG instead of SVG.NET.

Where did you get the DLLs you are writing about? In the latest SVG nuget package there are DLLs for several platforms: NETSTANDARD 2.0 and 2.1 (these should be usable for all environments that use that standard, including .NET framework), .NET 4.6.2 (.NET framework specific), and .NET Core 3.1 / .NET 6.0 (.NET (Core)-specific).
